includes/class-wc-google-analytics.php

```diff
@@ -482,10 +482,42 @@ public function add_to_cart() {
 		$parameters['label']    = "'" . esc_js( $product->get_sku() ? __( 'ID:', 'woocommerce-google-analytics-integration' ) . ' ' . $product->get_sku() : "#" . $product->get_id() ) . "'";
 
 		if ( ! $this->disable_tracking( $this->ga_enhanced_ecommerce_tracking_enabled ) ) {
+			// Update SKU and attributes when variation selection changes
+			wc_enqueue_js(
+				'var wc_ga_integration_product_sku,
+					wc_ga_integration_variation_sku,
+					wc_ga_integration_variation_attributes;
+
+				wc_ga_integration_product_sku = "' . esc_js( $product->get_sku() ) . '";
+
+				$(document).on( "found_variation", "form.cart", function( e, variation ) {
+					wc_ga_integration_variation_attributes = variation.attributes;
+					if ( "undefined" !== typeof variation.sku && 0 !== variation.sku.length ) {
+						wc_ga_integration_variation_sku = variation.sku;
+					} else {
+						wc_ga_integration_variation_sku = wc_ga_integration_product_sku;
+					}
+				});
+
+				function wc_ga_integration_format_variant() {
+					var variant = "";
+					for ( key in wc_ga_integration_variation_attributes ) {
+						variant += key.replace( "attribute_", "" ) + ": " + wc_ga_integration_variation_attributes[key] + ","
+					}
+					return variant.substring( 0, variant.length -1 )
+				}'
+			);
+
 			$item = "{";
-			$item .= "'id': '" . esc_js( $product->get_sku() ? $product->get_sku() : ( '#' . $product->get_id() ) ) . "',";
+			$item .= "'id': ( 0 !== wc_ga_integration_variation_sku.length ) ? ( wc_ga_integration_variation_sku ) : ( '#' + $('.woocommerce-variation-add-to-cart input[name=\"product_id\"]').val() ),";
```

Comment:
This check fails when we try this for a simple product, because we can't check the length of a variable that hasn't been defined yet. It would be better to initialise the variable as an empty string.

Reply:
As the formatted strings are now [coming from the backend](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/cf5ee3c1ef844a3af4ee6dbff6c937fef1dd8d5f/includes/class-wc-google-analytics.php#L488) there isn't an issue between simple and variable products anymore.